catlin/pkg/validator/task_validator.go

```diff
@@ -0,0 +1,52 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+
+	"github.com/tektoncd/plumbing/catlin/pkg/parser"
+)
+
+type TaskValidator struct {
+	log *zap.SugaredLogger
+	res *parser.Resource
+}
+
+var _ Validator = (*TaskValidator)(nil)
+
+func (t *TaskValidator) Validate() Result {
+
+	result := Result{}
+
+	res, err := t.res.ToType()
+	if err != nil {
+		result.Error("failed to decode task - %s", err)
+		return result
+	}
+
+	for _, step := range res.(*v1beta1.Task).Spec.Steps {
+		img := strings.FieldsFunc(step.Image, func(c rune) bool { return c == ':' })
+
+		if len(img) == 1 || img[1] == "latest" {
+			result.Error("Task image (%s) must be tagged with a specific version", step.Image)
+		}
```

Comment:
We should use a package that parse image reference, like `go-containerregistry` `pkg/name` package or docker's packages. This doesn't take digests into account for example (although this code would work)

Reply:
Good point! I didn't think of it at all! 🤗

Reply:
`digest` would not be mandatory right? the image should be tagged with a version and not `latest`. Is that correct?

Reply:
@SM43 that's correct. Using a digest, `foo@sha256:abc…` would work too. I think, the current code "works" with digests, but *by change*. Although it's not gonna change in the long-term future, using already existing like to parse/extra part of the image reference is a long-term viable thing to do (allow to validate more accurately the image reference, etc.)